tests/test_virtualenv.py

```diff
@@ -666,3 +666,36 @@ def test_license_builtin(clean_python):
     out = out_b.decode()
     assert not proc.returncode
     assert "Ian Bicking and Contributors" not in out
+
+
+@pytest.mark.skipif(
+    sys.platform != "darwin" or sys.version_info[:2] not in ((2, 7), (3, 5), (3, 6), (3, 7)),
+    reason="macOS specific test on python 2.7, 3.5, 3.6 & 3.7",
+)
+def test_dual_arch_macosx(tmpdir):
+    lookup = {
```

Comment:
this test is very heavyweight, I mean downloading the installer from upstream 🤔

Reply:
There's no CI with dual-architecture python install to my knowledge. If you know of one, please do tell.
That means for complete integration tests as the one implemented here, I don't know better.
Maybe lighter tests could be done by mocking things a bit.
Anyway, I think your next comment needs to be addressed first.